diff -urN libdhcp-1.20.org/Makefile libdhcp-1.20/Makefile
--- libdhcp-1.20.org/Makefile	2007-01-11 21:57:02.000000000 +0100
+++ libdhcp-1.20/Makefile	2007-02-20 20:57:28.442623000 +0100
@@ -27,7 +27,7 @@
       test_pump_dhcp.c test_pump_static.c test_dhcp_fork.c \
       Makefile.examples README COPYING
 
-export CFLAGS := $(CFLAGS) -fPIC -I. -Wall -Werror -g2 -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -D_GNU_SOURCE
+export CFLAGS := $(OPTFLAGS) -fPIC -I. -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -D_GNU_SOURCE
 
 DHC4_CFLAGS     = $(shell pkg-config --cflags libdhcp4client)
 DHC4_ISC_CFLAGS = $(shell pkg-config --variable=ISC.Cflags libdhcp4client)
